### PR TITLE
fix: display picture - prevent invalid files

### LINKED ISSUE #22592: [Feature]: Advanced file type validation is required on profile image upload
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

There is a basic client side validation exists for validating the file extension. But Users can alter the file extension to image and upload it as their profile image. 

### Why should this be worked on?

The application allows the attacker to upload or transfer files of dangerous types that can be automatically processed within the product's environment.During the assessment, it was observed that application allows to upload the php and js based files to the application which is restricted to upload the file.By changing the extension it is possible to upload the malicious file.

### PR DESCRIPTION
**Pull request by eco-monk**
## Description

When selecting a display picture, show alert on selecting invalid file type.
e.g. when you rename a pdf file to png and try to upload it

Design system changes: https://github.com/appsmithorg/design-system/pull/466

Fixes #22592

Media
https://www.loom.com/share/f8971769aaad422fb86c6b02306102ce

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Cypress
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/UserProfile/UserProfileImagePicker.tsx [hunk 1]
```diff
@@ -53,10 +56,24 @@ function FormDisplayImage() {
...
+      type: ReduxActionErrorTypes.USER_IMAGE_INVALID_FILE_CONTENT,
+      payload,
+    });
+  };
+
   return (
     <DisplayImageUpload
       disableUppyInformer={isAirgappedInstance}
       onChange={onSelectFile}
+      onInvalidFileContent={onFileTypeInvalid}
```
Comment (akash-codemonk) [inline app/client/src/pages/UserProfile/UserProfileImagePicker.tsx:76]: The changes for `design-system-old` might have to be added back to the new  `design-system` which might have a variation of this component. I think a cypress test might help to catch this missing implementation when the new `design-system` is added.

**Comment by Richarex** [comment]
Tested this on the DP url, working as expected user is informed with an error for invalid file format :- https://jiju8jbmwa.vmaker.com/record/t7XDpoSLuhsPzVkB

**Review by hetunandu — APPROVED** [review]

File: app/client/src/pages/UserProfile/UserProfileImagePicker.tsx (hunk 1 above)
Comment (eco-monk) [reply app/client/src/pages/UserProfile/UserProfileImagePicker.tsx:76]: done

**Comment by eco-monk** [comment]
Awaiting design system update.

**Review by akash-codemonk — DISMISSED** [review]

**Review by akash-codemonk — APPROVED** [review]

**Pull request merged**